### PR TITLE
Add CompositeModel hints test

### PR DESCRIPTION
**Pull request by tritemio**
* Add tests for hints propagation from base models to composite models
* Improve error messages for name collisions and non initialized parameters

### CODE REVIEW

File: tests/test_model.py [hunk 1]
```diff
@@ -344,6 +344,23 @@ def test_sum_composite_models(self):
...
+        m2 = Model(func, prefix='p2_')
+
+        m1.set_param_hint('amplitude', value=1)
+        m2.set_param_hint('amplitude', value=2)
+
+        mx = (m1 + m2)
+        params = mx.make_params()
+        param_values = {name: p.value for name, p in params.items()}
+        self.assertTrue(param_values['p1_amplitude'] == 1)
+        self.assertTrue(param_values['p2_amplitude'] == 2)
```
Comment (danielballan) [inline tests/test_model.py]: Just FYI, `self.assertEqual` would be better here. It gives a more useful error message when violated, printing the actual and expected values.

**Comment by danielballan** [comment]
I really like the improved error messages. This has been on my wishlist for awhile.

File: tests/test_model.py (hunk 1 above)
Comment (tritemio) [inline tests/test_model.py]: Good point, I will change it.

**Comment by tritemio** [comment]
Changed assertTrue with assertEqual.

In the meanwhile I noticed that [this other test](https://github.com/tritemio/lmfit-py/blob/eae879458a82802c168259d78a8e1f7fad8c9d2e/tests/test_model.py#L197) may be missing a final assertion?

**Comment by newville** [comment]
@trimetio @danielballan : This looks good.  I'd be willing to merge, but also willing to wait for that **other test** to be clarified.  Suggestions?

**Comment by tritemio** [comment]
The "other test" is an unrelated problem I spotted in the same file while I was adding this test. I opened a new issue #184 so it will not be forgotten.

**Comment by newville** [comment]
Great, thanks!

**Pull request merged**